service/frontend/wrappers/ratelimited/api_generated.go

```diff
@@ -0,0 +1,417 @@
+		return
+	}
+	return h.wrapped.ListClosedWorkflowExecutions(ctx, lp1)
+}
+
+func (h *apiHandler) ListDomains(ctx context.Context, lp1 *types.ListDomainsRequest) (lp2 *types.ListDomainsResponse, err error) {
+	return h.wrapped.ListDomains(ctx, lp1)
+}
+
+func (h *apiHandler) ListOpenWorkflowExecutions(ctx context.Context, lp1 *types.ListOpenWorkflowExecutionsRequest) (lp2 *types.ListOpenWorkflowExecutionsResponse, err error) {
+	if ok := h.allowDomain(ratelimitTypeVisibility, lp1.GetDomain()); !ok {
+		err = &types.ServiceBusyError{Message: "Too many outstanding requests to the cadence service"}
+		return
+	}
+	return h.wrapped.ListOpenWorkflowExecutions(ctx, lp1)
+}
+
+func (h *apiHandler) ListTaskListPartitions(ctx context.Context, lp1 *types.ListTaskListPartitionsRequest) (lp2 *types.ListTaskListPartitionsResponse, err error) {
+	if ok := h.allowDomain(ratelimitTypeUser, lp1.GetDomain()); !ok {
+		err = &types.ServiceBusyError{Message: "Too many outstanding requests to the cadence service"}
+		return
+	}
+	return h.wrapped.ListTaskListPartitions(ctx, lp1)
+}
+
+func (h *apiHandler) ListWorkflowExecutions(ctx context.Context, lp1 *types.ListWorkflowExecutionsRequest) (lp2 *types.ListWorkflowExecutionsResponse, err error) {
+	if ok := h.allowDomain(ratelimitTypeVisibility, lp1.GetDomain()); !ok {
+		err = &types.ServiceBusyError{Message: "Too many outstanding requests to the cadence service"}
+		return
+	}
+	return h.wrapped.ListWorkflowExecutions(ctx, lp1)
+}
+
+func (h *apiHandler) PollForActivityTask(ctx context.Context, pp1 *types.PollForActivityTaskRequest) (pp2 *types.PollForActivityTaskResponse, err error) {
+	if ok := h.allowDomain(ratelimitTypeWorker, pp1.GetDomain()); !ok {
+		err = &types.ServiceBusyError{Message: "Too many outstanding requests to the cadence service"}
+		return
+	}
+	return h.wrapped.PollForActivityTask(ctx, pp1)
+}
+
+func (h *apiHandler) PollForDecisionTask(ctx context.Context, pp1 *types.PollForDecisionTaskRequest) (pp2 *types.PollForDecisionTaskResponse, err error) {
+	if ok := h.allowDomain(ratelimitTypeWorker, pp1.GetDomain()); !ok {
+		err = &types.ServiceBusyError{Message: "Too many outstanding requests to the cadence service"}
+		return
+	}
+	return h.wrapped.PollForDecisionTask(ctx, pp1)
+}
+
+func (h *apiHandler) QueryWorkflow(ctx context.Context, qp1 *types.QueryWorkflowRequest) (qp2 *types.QueryWorkflowResponse, err error) {
+	if ok := h.allowDomain(ratelimitTypeUser, qp1.GetDomain()); !ok {
+		err = &types.ServiceBusyError{Message: "Too many outstanding requests to the cadence service"}
+		return
+	}
+	return h.wrapped.QueryWorkflow(ctx, qp1)
+}
+
+func (h *apiHandler) RecordActivityTaskHeartbeat(ctx context.Context, rp1 *types.RecordActivityTaskHeartbeatRequest) (rp2 *types.RecordActivityTaskHeartbeatResponse, err error) {
+	token, err := h.tokenSerializer.Deserialize(rp1.TaskToken)
+	if err != nil {
+		return
+	}
+	domainName, err := h.domainCache.GetDomainName(token.DomainID)
```

Comment:
probably missing an empty-check / the `validate.ErrDomainNotSet` branch.  the domain cache doesn't seem to check for empty values early on at least, kinda seems like it might not expect emptys.

we might also want to change the deserialize-err to a bad-request too, but technically that'd be a semantic change / only if you feel like it IMO.

Reply:
yea, domain cache will go this all the way down to the database without checking.  it's worth checking here.